main.c

```diff
@@ -58,7 +73,66 @@ void udelay (unsigned usec)
             break;
     }
 }
-
+/* 
+ * Read configuration registers value, its interpretation 
+ * and save it into the struct mips_cpuinfo
+ */
+void read_config()
+{
+    int config = mfc0(16, 0);
```

Comment:
Please use symbolic names for CP0 registers. Look at `mips-mti-elf/include/mips/m32c0.h` in the toolchain directory.